desk/app/channels-server.hoon

```diff
@@ -842,4 +1004,287 @@
+      (skip ids |=(i=id:h =(id i)))
+    =.  delayed.hooks
+      %+  roll
+        ~(tap by delayed.hooks)
+      |=  [[=delay-id:h d=[* delayed-hook:h]] dh=_delayed.hooks]
+      ?.  =(id hook.d)  dh
+      (~(del by dh) delay-id)
+    =.  cor  (give-hook-response [%gone id])
+    ho-core
+  ++  ho-configure
+    |=  [=nest:c =config:h]
+    ^+  ho-core
+    =.  config.hook  (~(put by config.hook) nest config)
+    =.  cor  (give-hook-response [%config id nest config])
+    ho-core
+  ++  ho-wait
+    |=  [=origin:h schedule=$@(@dr schedule:h) =config:h]
+    ^+  ho-core
+    =/  schedule
+      ?:  ?=(@ schedule)  [now.bowl schedule]
+      schedule
+    =/  crons  (~(gut by crons.hooks) id *(map origin:h cron:h))
+    =/  =cron:h  [id schedule config]
+    =.  crons.hooks
+      =-  (~(put by crons.hooks) id.hook -)
+      (~(put by crons) origin cron)
+    =.  cor  (schedule-cron origin cron)
+    =.  cor  (give-hook-response [%wait id origin schedule config])
+    ho-core
+  ++  ho-rest
+    |=  =origin:h
+    ^+  ho-core
+    =/  crons  (~(got by crons.hooks) id)
+    =/  cron  (~(got by crons) origin)
+    =.  crons.hooks
+      (~(put by crons.hooks) id (~(del by crons) origin))
+    =.  cor  (unschedule-cron origin cron)
+    =.  cor  (give-hook-response [%rest id origin])
+    ho-core
+  ++  ho-run-single
+    |=  [=event:h prefix=tape =origin:h =config:h]
+    =/  channel
+      ?@  origin  ~
+      ?~  ch=(~(get by v-channels) origin)  ~
+      `[origin u.ch]
+    =/  =context:h  (get-hook-context channel config)
+    =/  return=(unit return:h)
+      (run-hook:utils [event context(hook hook)] hook)
+    ?~  return
+      %-  (slog (crip "{prefix} {<id>} failed, running on {<origin>}") ~)
+      ho-core
+    %-  (slog (crip "{prefix} {<id>} ran on {<origin>}") ~)
+    =.  hook  hook(state new-state.u.return)
+    =.  cor  (run-hook-effects effects.u.return origin)
+    ho-core
+  --
+++  run-hooks
+  |=  [=event:h =nest:c default=cord]
+  ^-  [(each event:h tang) _cor]
+  =;  [result=(each event:h tang) effects=(list effect:h)]
+    [result (run-hook-effects effects nest)]
+  =/  current-event  event
+  =|  effects=(list effect:h)
+  =/  order  (~(got by order.hooks) nest)
+  =/  channel  `[nest (~(got by v-channels) nest)]
+  =/  =context:h  (get-hook-context channel *config:h)
+  |-
+  ?~  order
+    [&+current-event effects]
+  =*  next  $(order t.order)
+  =/  hook  (~(got by hooks.hooks) i.order)
+  =/  ctx  context(hook hook, config (~(gut by config.hook) nest ~))
+  =/  return=(unit return:h)
+    (run-hook:utils [current-event ctx] hook)
+  ?~  return  next
+  =*  result  result.u.return
+  =.  effects  (weld effects effects.u.return)
+  =.  hooks.hooks  (~(put by hooks.hooks) i.order hook(state new-state.u.return))
+  ?:  ?=(%denied -.result)
+    [|+~[(fall msg.result default)] effects]
+  =.  current-event  new.result
+  next
+++  wakeup-hook
+  |=  =(pole knot)
+  ^+  cor
+  ?+  pole  ~|(bad-arvo-take/pole !!)
```

Comment:
There is an unwritten convention to use `bad-arvo-take+pole` in such cases. In general I think we prefer `+` for a single use, `/` with a double tag use.